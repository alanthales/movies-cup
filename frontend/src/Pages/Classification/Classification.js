import React from "react";
import Button from "@material-ui/core/Button";
import ArrowBackIcon from "@material-ui/icons/ArrowBack"
import Layout from "../../Components/Layout/Layout";
import Rank from "../../Components/Rank/Rank";
import { useFetch } from "../../Components/Fetch/withFetch";

export const Classification = ({ history, location, data, isLoading, hasError }) => (
  <Layout
    title="Resultado Final"
    subtitle="Veja o resultado final do Campeonato de filmes de forma simples e rÃ¡pida."
  >
    <div style={{ marginBottom: "20px" }}>
      <Button variant="outlined" color="secondary" onClick={() => history.goBack()}>
        <ArrowBackIcon />
        Voltar
      </Button>
    </div>

    {isLoading && <p className="text-center">Carregando...</p>}

    {hasError && <p className="text-center">Opps! Algo deu errado, tente novamente.</p>}

    {!isLoading && !hasError && <Rank champion={data.campeao} vice={data.vice} />}
  </Layout>
);

const InjectedClassification = (props) => {
  const { movies } = props.location.state || [];
  const [data, isLoading, hasError] = useFetch("/api/movies/classification", {}, "post", movies);
  return (
    <Classification {...props} data={data} isLoading={isLoading} hasError={hasError} />
  );
};

export default InjectedClassification;