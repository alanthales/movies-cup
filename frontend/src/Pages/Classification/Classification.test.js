import React from "react";
import renderer from "react-test-renderer";
import { Classification } from "./Classification";

test("render Classification with isLoading prop", () => {
  const component = renderer.create(
    <Classification isLoading={true} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("render Classification with hasError prop", () => {
  const component = renderer.create(
    <Classification isLoading={false} hasError={true} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("render Classification with data prop", () => {
  const data = {
    campeao: { titulo: "Campeao"},
    vice: { titulo: "Vice" }
  };

  const component = renderer.create(
    <Classification isLoading={false} hasError={false} data={data} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});