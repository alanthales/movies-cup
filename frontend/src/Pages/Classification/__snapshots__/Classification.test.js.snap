// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Classification with data prop 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-146"
    >
      <div
        className="MuiGrid-root header-component MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <h4>
            CAMPEONATO DE FILMES
          </h4>
          <h1>
            Resultado Final
          </h1>
          <hr
            style={
              Object {
                "width": "50px",
              }
            }
          />
          <p>
            Veja o resultado final do Campeonato de filmes de forma simples e rápida.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Voltar
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root rank-component MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root champions MuiGrid-item"
      >
        <div
          className="MuiBox-root MuiBox-root-147 position"
        >
          1
          º
        </div>
        <div
          className="title"
        >
          <h4>
            Campeao
          </h4>
        </div>
      </div>
      <div
        className="MuiGrid-root champions MuiGrid-item"
      >
        <div
          className="MuiBox-root MuiBox-root-148 position"
        >
          2
          º
        </div>
        <div
          className="title"
        >
          <h4>
            Vice
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render Classification with hasError prop 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-145"
    >
      <div
        className="MuiGrid-root header-component MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <h4>
            CAMPEONATO DE FILMES
          </h4>
          <h1>
            Resultado Final
          </h1>
          <hr
            style={
              Object {
                "width": "50px",
              }
            }
          />
          <p>
            Veja o resultado final do Campeonato de filmes de forma simples e rápida.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Voltar
        </span>
      </button>
    </div>
    <p
      className="text-center"
    >
      Opps! Algo deu errado, tente novamente.
    </p>
  </div>
</div>
`;

exports[`render Classification with isLoading prop 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-105"
    >
      <div
        className="MuiGrid-root header-component MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <h4>
            CAMPEONATO DE FILMES
          </h4>
          <h1>
            Resultado Final
          </h1>
          <hr
            style={
              Object {
                "width": "50px",
              }
            }
          />
          <p>
            Veja o resultado final do Campeonato de filmes de forma simples e rápida.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Voltar
        </span>
      </button>
    </div>
    <p
      className="text-center"
    >
      Carregando...
    </p>
  </div>
</div>
`;
